text/0050-properties-window.md

```diff
@@ -0,0 +1,243 @@
+
+
+## Don't use checkable QGroupBox in the narrow layout
+
+For consistency with the check boxes above, instead of using checkable QGroupBoxes, I propose to display the group box name in a label and a checkbox in an accessory.
+
+Instead of:
+
+```
+._[x]_Group_name_____.
+|                    |
+|____________________|
+```
+show:
+```
+Group name         [x]
+.____________________.
+|                    |
+|____________________|
+```
+
+Although this adds a bit of height, it makes the group box name stand out more clearly, and in my code I have added a spacer above the group name to help each group appear separated from the one above it.
+
+# Examples
+
+Please note: these screenshots come from a WIP state where I have merged Properties and Filters into a single interface. I will propose this in a separate RFC.
+
+## Audio sources
+
+Note: since audio captures have no preview, it is hidden and the window is made to fit the only remaining contents, the OBSPropertiesView. This is a hint at what OBSBasicProperties could look like as a dock.
+
+| | |
+| :-: | :-: |
+| ![Audio Input Capture](0050-properties-window/audio-input-capture-source.png) | ![Audio Output Capture](0050-properties-window/audio-output-capture-source.png) |
+
+## Browser source
+
+![Browser source](0050-properties-window/browser-source.png)
+
+## Color source
+
+![Color source](0050-properties-window/color-source.png)
+
+## Image source
+
+![Image source](0050-properties-window/image-source.png)
+
+## Image slideshow source
+
+![Image slideshow](0050-properties-window/image-slideshow-source.png)
+
+## Media source
+
+![Media source](0050-properties-window/media-source.png)
+
+## Text source
+
+![Text source, single line](0050-properties-window/text-source-single-line.png)
+
+![Text source, multi-line](0050-properties-window/text-source-multi-line.png)
+
+## Video capture source
+
+![Video capture source](0050-properties-window/video-capture-source.png)
+
+## Effect filter
+
+![Effect filter](0050-properties-window/effect-filter.png)
+
+## Async filter
+
+![Async filter](0050-properties-window/async-filter.png)
+
+## Transition
+
+![Transition](0050-properties-window/transition.png)
+
+## Wide layout
+
+![Settings: Streaming](0050-properties-window/settings-streaming.png)
+
+![Settings: Recording](0050-properties-window/settings-recording.png)
+
+# Drawbacks
+
+From a design point of view, for this change to have any meaningful impact on the number of properties controls visible on screen at any given time, the following concessions must be made:
+
+- the font size must be reduced across the board to a consistently small-but-legible size
+  - this may harm accessibility
+  - that said, many of the included themes use 10pt or 11pt as a "small" size for certain controls
+  - in my code, I have chosen 11pt
+- the OBSPropertiesView is re-used by OBSBasicSettings to display certain controls which do not require a narrow layout
```

Comment:
Maybe it's because I'm tired (recently woke up) when I write those lines.

But I feel like `OBSPropertiesView` are considered in this RFC as something that is only used by source and filter which is wrong, Properties API is almost used by any object type provided by libobs (encoder, services, outputs…) even if not *actually* used, other RFC are here to change that.

Reply:
Properties **API** is used in lots of places, but OBSPropertiesView is only used by window-basic-properties (which includes the transitions window), window-basic-filters, window-basic-settings, AJAOutputUI, and DecklinkOutputUI.

I've already accommodated the fact that OBSPropertiesView is used in places other than the properties and filters with the wideLayout flag, which displays everything as it already is.

Reply:
`OBSPropertiesView` is for me bound to the Properties API, this is the type that interpret it.

Otherwise maybe it's just the wording "re-used" that lead me to that, but it just nitpicking at this level.

Reply:
> Edit: IIRC even the DeckLink UI use it.
> Edit 2: I forgot, scripting use it too.

And for that, my changes to OBSPropertiesView still supports the current layout.